src/lib/y2network/connection_config/usb.rb

```diff
@@ -0,0 +1,28 @@
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "y2network/connection_config/base"
+
+module Y2Network
+  module ConnectionConfig
+    # Configuration for Usb connections
+    class Usb < Base
```

Comment:
it is a bit strange that in reader/writter it is ethernet and here only base. So what if read want to write some eth specific attribute?

Reply:
It was mainly because of the `update_file` and `update_connection_config` methods. As there was not particular attributes in the USB connection, but once these methods have been just implemented in the Base class there is no reason to inherit from Ethernet anymore.